cmake/modules/AddSwift.cmake

```diff
@@ -343,12 +344,18 @@ endfunction()
 #
 # Usage:
 #   _add_swift_lipo_target(
+#     sdk                 # The name of the SDK the target was created for.
+#                         # Examples include "OSX", "IOS", "ANDROID", etc.
 #     target              # The name of the target to create
 #     output              # The file to be created by this target
 #     source_targets...   # The source targets whose outputs will be
 #                         # lipo'd into the output.
 #   )
-function(_add_swift_lipo_target target output)
+function(_add_swift_lipo_target sdk target output)
+  if("${sdk}" STREQUAL "")
+    message(FATAL_ERROR "sdk is required")
+  endif()
```

Comment:
I added some new utilities called precondition. It makes this a bit cleaner:

precondition("${sdk}" MESSAGE "sdk is required")